solana/solana-ibc/programs/solana-ibc/src/lib.rs

```diff
@@ -144,6 +146,23 @@ pub mod solana_ibc {
             .map_err(move |err| error!((&err)))
     }
 
+    pub fn deliver_with_chunks<'a, 'info>(
+        ctx: Context<'a, 'a, 'a, 'info, DeliverWithChunks<'info>>,
+    ) -> Result<()> {
+        let msg_chunks = &ctx.accounts.msg_chunks;
+        let cloned_msg_chunks: &mut MsgChunks = &mut msg_chunks.clone();
+        let mut store = storage::from_ctx!(ctx, with accounts);
+        let mut router = store.clone();
+
+        let message = ibc::MsgEnvelope::try_from(ibc::Any::from(
+            cloned_msg_chunks.clone(),
+        ))
```

Comment:
Why do we clone msg_chunks twice? Why is `cloned_msg_chunks` an exclusive reference?

Reply:
its a reference because i needed msg_chunks as `MsgChunks` so that i could convert it. But i think we can ignore it as we are doing the deserializing stuff in the `form_msg_chunks` method itself. So we would just be fetching the `MsgEnvelope` from the account and hence no conversions would be required in the `deliver_chunks` method.